src/token/index.js

```diff
@@ -16,7 +17,7 @@ const validateSymbol = (symbol)=>{
   }
 }
 
-const validateAmount = async (amount, symbol, fromAddress, httpClient, type="free") => {
+const validateNonZeroAmount = async (amount, symbol, fromAddress, httpClient, type="free") => {
   if(amount <= 0 || amount > MAXTOTALSUPPLY){
     throw new Error("invalid amount")
```

Comment:
Again, this should not really throw an error here but should instead return `true` or `false` and let the caller throw the error.